### PR TITLE
Add setMac arg to Wiz5500 constructor to support WIZ550io

### PR DESCRIPTION
**Pull request by chrisjbremner**
@ppetrosh 

I am developing with a [WIZ550io](http://www.wiznet.io/product-item/wiz550io/) board which comes with a built-in MAC address. Because of this, I do not want `W5500.Driver.init` to manually set the MAC address to that of the imp itself. This PR adds as `setMac` optional argument to the constructor of the `W5500` class which allows you to turn off this MAC setting functionality if you want.

I've tested this with my own WIZ550io device. Untested on a stand-alone W5500 chip since I don't have one, but it should not affect things.

### CODE REVIEW

**Comment by ppetrosh** [comment]
Thanks @chrisjbremner! The change looks good to me. Could you please pump up the version to 2.1.0 in the source code, documentation and examples? Thanks.

**Comment by chrisjbremner** [comment]
Thanks for the speedy review @ppetrosh, I have bumped the version in the source, documentation, examples, and a test. I also added the new parameter to the README since I hadn't done that. Let me know if you see anything else.

File: W5500.device.lib.nut [hunk 1]
```diff
@@ -526,13 +528,15 @@ class W5500.Driver {
     //      spi - configured spi, W5500 supports spi mode 0 or 3
     //      cs(optional) - configured chip select pin
     //      reset(optional) - configured reset pin
+    //      setMac(optional) - set the MAC address of the chip to the imp's MAC
     // ***************************************************************************
-    constructor(interruptPin, spi, cs = null, resetPin = null) {
+    constructor(interruptPin, spi, cs = null, resetPin = null, setMac = false) {
```
Comment (ppetrosh) [inline W5500.device.lib.nut]: The default behavior should not change so I believe this should default to true.

Comment (chrisjbremner) [reply W5500.device.lib.nut]: Oops, missed that one. Fixed in https://github.com/electricimp/Wiznet_5500/pull/10/commits/f6c8b006e40f1e763982f009f32f21c294854a5c

**Review by ppetrosh — APPROVED** [review]

**Comment by ppetrosh** [comment]
Thank you @chrisjbremner for addressing the comments!

**Pull request merged**